src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -99,76 +99,85 @@ private static void WriteClientFields(CodeWriter writer, LowLevelClient client)
 
-            if (client.IsSubClient)
+            foreach (var constructor in client.PrimaryConstructors)
             {
-                WriteSubClientInternalConstructor(writer, client, client.SubClientInternalConstructor);
+                WritePublicConstructor(writer, client, constructor, true);
             }
-        }
 
-        private static void WriteEmptyConstructor(CodeWriter writer, TypeProvider client)
-        {
-            writer.WriteXmlDocumentationSummary($"Initializes a new instance of {client.Type.Name} for mocking.");
-            using (writer.Scope($"protected {client.Type.Name:D}()"))
+            if (client.IsSubClient)
             {
+                WriteSubClientInternalConstructor(writer, client, client.SubClientInternalConstructor);
             }
-            writer.Line();
         }
 
-        private static void WritePublicConstructor(CodeWriter writer, LowLevelClient client, ConstructorSignature signature)
+        private static void WritePublicConstructor(CodeWriter writer, LowLevelClient client, ConstructorSignature signature, bool isPrimary)
         {
             writer.WriteMethodDocumentation(signature);
             using (writer.WriteMethodDeclaration(signature))
             {
-                writer.WriteParametersValidation(signature.Parameters);
-                writer.Line();
+                if (isPrimary)
+                {
+                    writer.WriteParametersValidation(signature.Parameters);
+                    writer.Line();
 
-                var clientOptionsParameter = signature.Parameters.Last(p => p.Type.EqualsIgnoreNullable(client.ClientOptions.Type));
-                writer.Line($"{client.Fields.ClientDiagnosticsProperty.Name:I} = new {client.Fields.ClientDiagnosticsProperty.Type}({clientOptionsParameter.Name:I});");
+                    var clientOptionsParameter = signature.Parameters.Last(p => p.Type.EqualsIgnoreNullable(client.ClientOptions.Type));
+                    writer.Line($"{client.Fields.ClientDiagnosticsProperty.Name:I} = new {client.Fields.ClientDiagnosticsProperty.Type}({clientOptionsParameter.Name:I});");
 
-                FormattableString perCallPolicies = $"Array.Empty<{typeof(HttpPipelinePolicy)}>()";
-                FormattableString perRetryPolicies = $"Array.Empty<{typeof(HttpPipelinePolicy)}>()";
+                    FormattableString perCallPolicies = $"Array.Empty<{typeof(HttpPipelinePolicy)}>()";
+                    FormattableString perRetryPolicies = $"Array.Empty<{typeof(HttpPipelinePolicy)}>()";
 
-                var credentialParameter = signature.Parameters.FirstOrDefault(p => p.Name == "credential");
-                if (credentialParameter != null)
-                {
-                    var credentialField = client.Fields.GetFieldByParameter(credentialParameter);
-                    if (credentialField != null)
+                    var credentialParameter = signature.Parameters.FirstOrDefault(p => p.Name == "credential");
```

Comment:
I may be wrong, but I think it can be simplified:
```cs
foreach (var parameter in signature.Parameters)
{
    var field = client.Fields.GetFieldByParameter(parameter);
    if (field != null)
    {
        if (field.Type.Equals(typeof(AzureKeyCredential)))
        {
            writer.Line($"{field.Name:I} = {parameter.Name:I};");
            perRetryPolicies = $"new {typeof(HttpPipelinePolicy)}[] {{new {typeof(AzureKeyCredentialPolicy)}({field.Name:I}, {client.Fields.AuthorizationHeaderConstant!.Name})}}";
        }
        else if (field.Type.Equals(typeof(TokenCredential)))
        {
            writer.Line($"{field.Name:I} = {parameter.Name:I};");
            perRetryPolicies = $"new {typeof(HttpPipelinePolicy)}[] {{new {typeof(BearerTokenAuthenticationPolicy)}({field.Name:I}, {client.Fields.ScopesConstant!.Name})}}";
        }
        else if (parameter.IsApiVersionParameter)
        {
            writer.Line($"{field.Name:I} = {clientOptionsParameter.Name:I}.Version;");
        }
        else
        {
            writer.Line($"{field.Name:I} = {parameter.Name:I};");
        }
    }
}

writer.Line($"{client.Fields.PipelineField.Name:I} = {typeof(HttpPipelineBuilder)}.{nameof(HttpPipelineBuilder.Build)}({clientOptionsParameter.Name:I}, {perCallPolicies}, {perRetryPolicies}, new {typeof(ResponseClassifier)}());");
```

Reply:
In this case, we only go through the signature parameters, but we still have client parameters. And they have overlap. E.g., clinet parameter might contains apiversion parameter, while signature not.

Reply:
Ok, let's leave it as it is for now